adapters/adnuntius/adnuntius.go

```diff
@@ -315,6 +158,96 @@ func (a *adapter) generateRequests(ortbRequest openrtb2.BidRequest) ([]*adapters
 	return requestData, nil
 }
 
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	return a.generateRequests(*request)
+}
+
+func generateBidResponse(adnResponse *AdnResponse, request *openrtb2.BidRequest) (*adapters.BidderResponse, []error) {
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(adnResponse.AdUnits))
+	var currency string
+	adunitMap := map[string]AdUnit{}
+	adunitMediaTypeMap := map[string][]AdUnit{}
+
+	/* Check the ad unit response to see if there are any multi ad  */
+	for _, adnRespAdunit := range adnResponse.AdUnits {
+		result := strings.Split(adnRespAdunit.TargetId, ":")
+		if adnRespAdunit.MatchedAdCount > 0 {
+			adunitMediaTypeMap[result[0]] = append(adunitMediaTypeMap[result[0]], adnRespAdunit)
+		}
+	}
+
+	/* Compare price if there are multiple media types */
+	for targetId, mappedAdunit := range adunitMediaTypeMap {
+		highestBidAtIndex := 0
+		if len(mappedAdunit) > 1 {
+			for index := range mappedAdunit {
+				if mappedAdunit[index].Ads[0].Bid.Amount > mappedAdunit[highestBidAtIndex].Ads[0].Bid.Amount {
+					highestBidAtIndex = index
+				}
+			}
+		}
+		adunitMap[targetId] = mappedAdunit[highestBidAtIndex]
+	}
+
+	for _, imp := range request.Imp {
+
+		auId, _, _, err := jsonparser.Get(imp.Ext, "bidder", "auId")
+		if err != nil {
+			return nil, []error{&errortypes.BadInput{
+				Message: fmt.Sprintf("Error at Bidder auId: %s", err.Error()),
+			}}
+		}
+
+		targetID := fmt.Sprintf("%s-%s", string(auId), imp.ID)
+
+		adunit := adunitMap[targetID]
+
+		if len(adunit.Ads) > 0 {
+
+			ad := adunit.Ads[0]
+			html := adunit.Html
+			var bidType openrtb2.MarkupType = 1
+
+			currency = ad.Bid.Currency
+			native, _, _, _ := jsonparser.Get(adunit.NativeJson, "ortb")
+
+			if native != nil {
+				html = string(native)
+				bidType = 4
+			}
+
+			adBid, err := generateAdResponse(ad, imp, html, bidType, request)
+			if err != nil {
+				return nil, []error{&errortypes.BadInput{
```

Comment:
Here we are ignoring the original error `err` and returning a new error under the `BadInput` error type. Is this the intended behaviour? 

Additionally, in the `generateAdResponse` method, many errors are caused by issues in the bidder response, such as an invalid ad.creativeWidth. It seems appropriate for these errors to fall under the BadServerResponse error type. Should we align them accordingly?

Reply:
Also, please add supplemental JSON test cases to cover these scenario

Reply:
I found it hard to create a test for this since all error scenarios makes it throw an error earlier in the code that's already tested for.